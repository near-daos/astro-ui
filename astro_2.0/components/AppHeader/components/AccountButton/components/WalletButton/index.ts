export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
