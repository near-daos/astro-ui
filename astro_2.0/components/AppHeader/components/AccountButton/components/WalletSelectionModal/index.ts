export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
