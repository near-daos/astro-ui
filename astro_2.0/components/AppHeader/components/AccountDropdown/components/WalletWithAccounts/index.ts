export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
