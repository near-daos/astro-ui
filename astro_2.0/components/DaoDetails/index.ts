export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
