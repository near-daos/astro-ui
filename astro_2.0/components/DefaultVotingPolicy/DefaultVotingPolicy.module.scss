@import 'styles/mixins';
@import 'styles/typography';

.title {
  @extend %title5;
}

.policyWrapper {
  background-color: var(--color-brand-black);
  border-radius: 8px;
  margin-top: 56px;
  padding: 14px;

  @include tablet-portrait {
    padding: 14px 60px;
  }

  .policyLabel {
    @extend %title5;
    color: var(--color-white);
    margin: 0 0 12px;
  }

  .policy {
    align-items: center;
    color: var(--color-white);
    display: flex;
    justify-content: flex-start;
  }

  .policy > div {
    margin-right: 6px;
  }

  .bold {
    flex-shrink: 0;
    font-weight: bold;
  }
}
