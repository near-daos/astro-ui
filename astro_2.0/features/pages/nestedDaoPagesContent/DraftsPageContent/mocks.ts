import { ProposalType } from 'types/proposal';
import { DraftComment, DraftProposal } from 'types/draftProposal';

const proposal = {
  voteYes: 0,
  voteNo: 0,
  voteRemove: 0,
  votes: {},
  id: 'test.sputnikv2.testnet-6',
  proposalId: 6,
  daoId: 'test.sputnikv2.testnet',
  proposer: 'terratest.testnet',
  commentsCount: 0,
  description:
    'create multicall instance for this DAO at test.v1_02.multicall.testnet',
  link: '',
  status: 'InProgress',
  kind: {
    type: 'FunctionCall',
    receiverId: 'v1_02.multicall.testnet',
    actions: [
      {
        methodName: 'create',
        args:
          'eyJtdWx0aWNhbGxfaW5pdF9hcmdzIjp7ImFkbWluX2FjY291bnRzIjpbInRlc3Quc3B1dG5pa3YyLnRlc3RuZXQiXSwiY3JvbmNhdF9tYW5hZ2VyIjoibWFuYWdlcl92MS5jcm9uY2F0LnRlc3RuZXQiLCJqb2JfYm9uZCI6IjEwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAifX0=',
        deposit: '1000000000000000000000000',
        gas: '150000000000000',
      },
    ],
  },
  votePeriodEnd: '2022-06-03T02:21:05.837Z',
  votePeriodEndDate: '2022-06-03T02:21:05.837Z',
  voteStatus: 'Active',
  isFinalized: false,
  txHash: '6CfB5TM3iS9nQNAusXsqSSKhK2r9t3jfiKD4eRyHbiPr',
  createdAt: '2022-05-27T02:21:17.140Z',
  dao: {
    id: 'test.sputnikv2.testnet',
    name: 'test',
    logo: '/flags/defaultDaoFlag.png',
    flagCover: '',
    flagLogo: '',
    legal: {},
    numberOfMembers: 2,
    policy: {
      daoId: 'test.sputnikv2.testnet',
      defaultVotePolicy: {
        weightKind: 'RoleWeight',
        quorum: '0',
        kind: 'Ratio',
        ratio: [1, 2],
      },
    },
  },
  daoDetails: {
    name: 'test',
    displayName: '',
    logo: '/flags/defaultDaoFlag.png',
  },
  proposalVariant: 'ProposeDefault',
  updatedAt: '2022-05-27T02:21:17.140Z',
  actions: [
    {
      id: 'test.sputnikv2.testnet-6-terratest.testnet-AddProposal',
      proposalId: 'test.sputnikv2.testnet-6',
      accountId: 'terratest.testnet',
      action: 'AddProposal',
      transactionHash: '6CfB5TM3iS9nQNAusXsqSSKhK2r9t3jfiKD4eRyHbiPr',
      timestamp: '1653618065837431071',
    },
  ],
  permissions: {
    isCouncil: false,
    canApprove: false,
    canReject: false,
    canDelete: false,
  },
};

const comment: DraftComment = {
  createdAt: '2021-12-26T20:30:46.036Z',
  updatedAt: '2021-12-26T20:30:46.036Z',
  id: '0',
  likes: 10,
  author: 'John Doe',
  description: '<p>Test</p>',
};

export const mocks = {
  count: 2,
  total: 2,
  page: 0,
  pageCount: 1,
  data: [
    {
      ...proposal,
      id: '0',
      type: ProposalType.Transfer,
      title: 'Draft Name kickstart medical outreaches by near medical',
      text: 'some desc here',
      hashtags: ['#mintbase', '#community', '#marketing'],
      views: 0,
      replies: 0,
      createdAt: '2021-12-26T20:30:46.036Z',
      updatedAt: '2021-12-26T20:30:46.036Z',
      isRead: false,
      isSaved: false,
      bookmarks: 10,
      comments: [comment],
      state: 'open',
    },
    {
      ...proposal,
      id: '1',
      type: ProposalType.FunctionCall,
      title: 'Some title',
      text: 'some desc here',
      hashtags: ['#mintbase', '#community', '#marketing'],
      views: 0,
      replies: 0,
      isRead: false,
      isSaved: false,
      bookmarks: 10,
      comments: [comment],
      state: 'open',
    },
  ] as DraftProposal[],
};
