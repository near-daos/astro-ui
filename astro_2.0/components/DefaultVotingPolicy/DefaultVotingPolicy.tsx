import React, { FC } from 'react';
import isArray from 'lodash/isArray';
import isEmpty from 'lodash/isEmpty';
import { useTranslation } from 'next-i18next';

import { Badge } from 'components/Badge';

import styles from './DefaultVotingPolicy.module.scss';

interface DefaultVotingPolicyProps {
  numberOfGroups: number;
  ratio: number[];
}

export const DefaultVotingPolicy: FC<DefaultVotingPolicyProps> = ({
  ratio,
  numberOfGroups,
}) => {
  const { t } = useTranslation();

  const amount =
    isArray(ratio) && !isEmpty(ratio) ? (ratio[0] / ratio[1]) * 100 : '';
  const threshold = 'of group';

  return (
    <div className={styles.policyWrapper}>
      <div className={styles.policyLabel}>{t('votingPolicy')}</div>
      <div className={styles.policy}>
        <div className={styles.bold}>{amount}%</div>
        <div>{threshold} to pass of</div>
        <Badge size="small" variant="primary">
          ALL GROUPS ({numberOfGroups})
        </Badge>
      </div>
    </div>
  );
};
