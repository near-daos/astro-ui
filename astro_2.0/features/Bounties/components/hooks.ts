import { useCallback, useRef, useState } from 'react';
import { useRouter } from 'next/router';
import { useAsyncFn, useLocalStorage, useLocation } from 'react-use';
import axios, { CancelTokenSource } from 'axios';

import { SputnikHttpService } from 'services/sputnik';
import { useModal } from 'components/modal';
import { ConfirmActionModal } from 'astro_2.0/components/ConfirmActionModal';

import { Bounty, BountyContext, BountyProposal } from 'types/bounties';
import { VoteAction } from 'types/proposal';

import { NOTIFICATION_TYPES, showNotification } from 'features/notifications';

import { useWalletContext } from 'context/WalletContext';
import { PaginationResponse } from 'types/api';
import { VOTE_ACTION_SOURCE_PAGE } from 'constants/votingConstants';

export function useBountyControls(
  daoId: string,
  bounty?: Bounty
): {
  handleClaim: () => void;
  handleUnclaim: () => void;
  loading: boolean;
} {
  const router = useRouter();
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();
  const [loading, setLoading] = useState(false);

  const [showModal] = useModal(ConfirmActionModal);

  const onSuccessHandler = useCallback(async () => {
    router.reload();
  }, [router]);

  const handleClaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Claim',
    });

    if (res?.length && bounty) {
      setLoading(true);

      const dao = await SputnikHttpService.getDaoById(daoId);

      if (!dao) {
        return;
      }

      try {
        setVoteActionSource(pathname);

        await nearService?.claimBounty(daoId, {
          bountyId: bounty?.bountyId,
          deadline: bounty?.maxDeadline,
          bountyBond: dao.policy.bountyBond,
          gas: res[0],
          tokenId: bounty?.token,
        });

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    daoId,
    setVoteActionSource,
    pathname,
    nearService,
    onSuccessHandler,
  ]);

  const handleUnclaim = useCallback(async () => {
    const res = await showModal({
      title: 'Confirm Your Unclaim',
    });

    if (res?.length && bounty) {
      try {
        setLoading(true);
        setVoteActionSource(pathname);
        await nearService?.unclaimBounty(daoId, bounty?.bountyId);

        await onSuccessHandler();
      } catch (err) {
        setLoading(false);
        showNotification({
          type: NOTIFICATION_TYPES.ERROR,
          lifetime: 5000,
          description: err.message,
        });
      }
    }
  }, [
    showModal,
    bounty,
    setVoteActionSource,
    pathname,
    nearService,
    daoId,
    onSuccessHandler,
  ]);

  return {
    handleClaim,
    handleUnclaim,
    loading,
  };
}

export function useBountyVoting(
  daoId: string,
  proposal: BountyProposal
): {
  handleVote: (vote: VoteAction) => void;
  loading: boolean;
} {
  const router = useRouter();
  const [showModal] = useModal(ConfirmActionModal);
  const { nearService } = useWalletContext();
  const [, setVoteActionSource] = useLocalStorage(VOTE_ACTION_SOURCE_PAGE);
  const { pathname } = useLocation();

  const [{ loading }, handleVote] = useAsyncFn(
    async (vote: VoteAction) => {
      const res = await showModal({
        title: 'Confirm Your Vote',
      });

      if (res?.length) {
        setVoteActionSource(pathname);
        await nearService?.vote(daoId, proposal.proposalId, vote, res[0]);
        await router.reload();
      }
    },
    [daoId, proposal, router, nearService, pathname, setVoteActionSource]
  );

  return {
    handleVote,
    loading,
  };
}

export function useBountySearch(): {
  handleSearch: (
    val: string
  ) => Promise<PaginationResponse<BountyContext[]> | null>;
  loading: boolean;
} {
  const router = useRouter();
  const daoId = router.query.dao as string;
  const { accountId } = useWalletContext();
  const cancelTokenRef = useRef<CancelTokenSource | null>(null);

  const [{ loading }, handleSearch] = useAsyncFn(async query => {
    if (cancelTokenRef.current) {
      cancelTokenRef.current?.cancel('Cancelled by new req');
    }

    const { CancelToken } = axios;
    const source = CancelToken.source();

    cancelTokenRef.current = source;

    return SputnikHttpService.findBountyContext({
      daoId: daoId || '',
      accountId,
      query,
      cancelToken: source.token,
    });
  }, []);

  return {
    handleSearch,
    loading,
  };
}

export function useHideBounty(): {
  handleSubmit: () => void;
  handleSelect: (id: string) => void;
  selected: string[];
  loading: boolean;
} {
  const router = useRouter();
  const showHidden = router.query?.bountyFilter === 'hidden';
  const daoId = router.query.dao as string;
  const { accountId, nearService, pkAndSignature } = useWalletContext();
  const [selected, setSelected] = useState<string[]>([]);

  const [{ loading }, handleSubmit] = useAsyncFn(async () => {
    if (!pkAndSignature) {
      return;
    }

    try {
      const { publicKey, signature } = pkAndSignature;

      if (publicKey && signature) {
        await SputnikHttpService.toggleBountyContexts({
          accountId,
          publicKey,
          signature,
          daoId,
          ids: selected,
          isArchived: !showHidden,
        });

        await router.replace(router.asPath, undefined, {
          shallow: false,
        });

        setSelected([]);
      }
    } catch (err) {
      showNotification({
        type: NOTIFICATION_TYPES.ERROR,
        lifetime: 20000,
        description: err.message,
      });
    }
  }, [
    selected,
    daoId,
    showHidden,
    accountId,
    router,
    nearService,
    pkAndSignature,
  ]);

  const handleSelect = useCallback(
    (id: string) => {
      if (selected.includes(id)) {
        setSelected(selected.filter(item => item !== id));
      } else {
        setSelected([...selected, id]);
      }
    },
    [selected]
  );

  return {
    handleSubmit,
    handleSelect,
    selected,
    loading,
  };
}
