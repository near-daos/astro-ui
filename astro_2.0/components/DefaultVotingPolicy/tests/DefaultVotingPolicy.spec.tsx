import { render } from 'jest/testUtils';

import { DefaultVotingPolicy } from 'astro_2.0/components/DefaultVotingPolicy';

import { policyMock } from './mocks';

describe('default voting policy', () => {
  it('Should render component', () => {
    const { container } = render(
      <DefaultVotingPolicy ratio={[]} numberOfGroups={2} />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render ration if ration info presented', () => {
    const { queryAllByText } = render(
      <DefaultVotingPolicy ratio={policyMock.ratio} numberOfGroups={2} />
    );

    expect(queryAllByText('50%', { exact: false })).toHaveLength(1);
  });
});
