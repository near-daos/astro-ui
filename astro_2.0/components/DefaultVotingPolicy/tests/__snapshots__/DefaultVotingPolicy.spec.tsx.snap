// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default voting policy Should render component 1`] = `
<div>
  <div
    class="policyWrapper"
  >
    <div
      class="policyLabel"
    >
      votingPolicy
    </div>
    <div
      class="policy"
    >
      <div
        class="bold"
      >
        
        %
      </div>
      <div>
        of group
         to pass of
      </div>
      <div
        class="badge small primary"
      >
        ALL GROUPS (
        2
        )
      </div>
    </div>
  </div>
</div>
`;
